domdiv/draw.py

```diff
@@ -62,167 +385,367 @@ def registerFonts(self):
-                    last_y = last_y + y
-            return result
-
-        self.canvas.saveState()
+            lineType = plotter.NO_LINE
+            lineTypeNoDot = plotter.NO_LINE
+
+        # Setup bare minimum lineStyle's
+        lineStyle = [lineType for i in range(0, 10)]
+        lineStyle[0] = lineTypeNoDot
+        lineStyle[7] = lineType
+        lineStyle[8] = lineType if left2tab > 0 else lineTypeNoDot
+        lineStyle[9] = lineType if right2tab > 0 else lineTypeNoDot
+
+        CropRight = self.options.cropmarks and item.translateCropmarkEnable(item.RIGHT)
+        CropLeft = self.options.cropmarks and item.translateCropmarkEnable(item.LEFT)
+        CropTop = self.options.cropmarks and item.translateCropmarkEnable(item.TOP)
+        CropBottom = self.options.cropmarks and item.translateCropmarkEnable(item.BOTTOM)
+
+        RIGHT = plotter.RIGHT
+        LEFT = plotter.LEFT
+        BOTTOM = plotter.BOTTOM
+        TOP = plotter.TOP
+        NO_LINE = plotter.NO_LINE
 
         if not self.options.wrapper:
             # Normal Card Outline
-            #    +                      F+-------------------+E
-            #                            |                   |
-            #   H+-----------------------+G                 D+-----+C
-            #    |                                                 |
-            #    |             Generic Divider                     |
-            #    |          Tab Centered or to the Side            |
-            #    |                                                 |
-            #   A+-------------------------------------------------+B
-            #             delta x          delta y
-            delta = [(0, 0),  # to A
-                     (dividerWidth, 0),  # A to B
-                     (0, dividerBaseHeight),  # B to C
-                     (-side_2_tab, 0),  # C to D
-                     (0, theTabHeight),  # D to E
-                     (-theTabWidth, 0),  # E to F
-                     (0, -theTabHeight),  # F to G
-                     (-nonTabWidth, 0),  # G to H
-                     (0, -dividerBaseHeight)]  # H to A
-            self.canvas.lines(DeltaXYtoLines(delta))
+            #     <-left2tab-> <--tabLabelWidth--> <-right2tab->
+            #    |            |                   |             |
+            #  Z-+           F7-------------------7E            +-Y
+            #                 |                   |
+            #  H-8------------8                   9-------------9-C
+            #    |            G                   D             |
+            #    |               Generic Divider                |
+            #    |            Tab Centered or to the Side       |
+            #    |                                              |
+            #  A-7------------0-------------------0-------------7-B
+            #    |           V|                  W|             |
+            #
+            plotter.move(0, 0)  # to A
```

Comment:
While I agree that it's somewhat more readable in terms of calculations, this has gotten very verbose.

Could we maybe
1) store the basic cropmark enabled flags in the plotter object
2) combine the `move` and all the `cropmark` calls into one `plot` call that takes the `destination point`, the `linestyle` (which could maybe be a bit more readable with something enum/dict like than the array indexes here) and a list of `cropmarks` to check whether to draw?

So like
`plotter.plot(right2tab, 0, lineStyle[7], [BOTTOM, RIGHT])`
for more complicated cases, the cropmark list could be preconstructed in case filtering is needed.